src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -429,4 +442,27 @@ trait WorkflowComponent {
 
   }
 
+  object workflowAuditStatusQuery extends TableQuery(new WorkflowAuditStatusTable(_)) {
+
+    def queueTimeMostRecentSubmittedWorkflow: ReadAction[Long] = {
+      // for the most-recently submitted workflow, find the time it was submitted and the earliest time we have recorded.
+      // we could query specifically for the time this workflow was Queued, but we'll just use the earliest time
+      // for resiliency, in case the workflow somehow skipped Queued status.
+      val timingQuery = for {
+        lastSubmitted <- workflowAuditStatusQuery.filter(_.status === WorkflowStatuses.Submitted.toString).sortBy(_.timestamp.desc).take(1)
+        earliestSeen <- workflowAuditStatusQuery.sortBy(_.timestamp.asc).take(1) if earliestSeen.workflowId === lastSubmitted.workflowId
+      } yield (lastSubmitted.timestamp, earliestSeen.timestamp)
+
+      timingQuery.result.headOption map {
+        case Some((s:Timestamp, e:Timestamp)) => s.getTime - e.getTime
+        case _ => 0L
+      }
+    }
+
+    def save(wasr: WorkflowAuditStatusRecord): WriteAction[WorkflowAuditStatusRecord] = {
```

Comment:
Make a note that this is for testing only, and writing to this table is done by a trigger.